shell/src/main/java/alluxio/cli/fs/command/DuCommand.java

```diff
@@ -171,17 +229,29 @@ private static String getFormattedValues(boolean readable, long size, long total
    * @param inAlluxioMessage the in Alluxio size message to print
    * @param inMemMessage the in memory size message to print
    * @param path the path to print
+   * @param workerHostName the worker host name to print
    */
-  private static void printInfo(String sizeMessage,
-      String inAlluxioMessage, String inMemMessage, String path) {
-    System.out.println(inMemMessage.isEmpty()
-        ? String.format(SHORT_INFO_FORMAT, sizeMessage, inAlluxioMessage, path)
-        : String.format(LONG_INFO_FORMAT, sizeMessage, inAlluxioMessage, inMemMessage, path));
+  private static void printInfo(String sizeMessage, String inAlluxioMessage,
+      Optional<String> inMemMessage, String path, Optional<String> workerHostName) {
+    String message;
+    if (inMemMessage.isPresent() && workerHostName.isPresent()) {
+      message = String.format(LONG_INFO_FORMAT, sizeMessage, inAlluxioMessage,
+              inMemMessage.get(), workerHostName.get(), path);
+    } else if (inMemMessage.isPresent()) {
+      message = String.format(MID_1_INFO_FORMAT, sizeMessage, inAlluxioMessage,
+              inMemMessage.get(), path);
+    } else if (workerHostName.isPresent()) {
+      message = String.format(MID_2_INFO_FORMAT, sizeMessage, inAlluxioMessage,
+              workerHostName.get(), path);
+    } else {
+      message = String.format(SHORT_INFO_FORMAT, sizeMessage, inAlluxioMessage, path);
+    }
+    System.out.println(message);
```

Comment:
There should be multiple ways to make this better:
1. Generate the FORMAT string based on whether the optional are present, then `String.format(generatedFormat, obj[])` give it the format and the list of strings. You can decide how many items in the obj[] based on whether they are present.

2. Also generate the format, with 5 elements, but if one element (for example `inMemMessage.equals("")`) then you can change the length for that part to maybe 0 or 1?

3. Up to you if you have better ideas.